assigner.py

```diff
@@ -20,173 +22,257 @@
-
-                    if args.branch:
+                if force:
+                    logging.warning("Deleting...") if verbose else None
+                    if not dry_run:
+                        repo.delete()
+                        repo = StudentRepo.new(base, semester, s_section, username, token)
                         repo.push(base, args.branch)
-                    else:
-                        repo.push(base)
-
-                    count += 1
+                    actual_count += 1
                 else:
                     # If we have an explicit branch, push anyways
-                    if args.branch:
-                        repo.push(base, args.branch)
-                        count += 1
+                    if branch:
+                        if not dry_run:
+                            repo.push(base, args.branch)
+                        actual_count += 1
                     else:
-                        logging.warning("Skipping...")
+                        logging.warning("Skipping...") if verbose else None
 
             except HTTPError as e:
                 if e.response.status_code == 404:
-                    repo = StudentRepo.new(base, conf['semester'], student['section'], student['username'], conf['token'])
-                    if args.branch:
-                        repo.push(base, args.branch)
-                    else:
-                        repo.push(base)
-                    count += 1
+                    if not dry_run:
+                        repo = StudentRepo.new(base, semester, s_section, username, token)
+                        repo.push(base, branch)
+                    actual_count += 1
                 else:
                     raise
+            i += 1
+            print_progress(i / s_count) if not verbose else None
 
-    print("Assigned homework ", args.name, " to ", count, " students")
+    print("{}Managed to assign homework {} to {} students.".format(
+        "\n" if not verbose else None, hw_name, actual_count)
+    )
+    if actual_count == 0:
+        print("Consider using --force if you want to override existing repos.")
 
 
 def open_assignment(args):
+    """Adds each student in the roster to their respective homework
+    repositories as Developers so they can pull/commit/push their work.
+    """
+    hw_name = args.name
+    section = args.section
+
     with config(args.config) as conf:
+        host = conf['gitlab-host']
+        namespace = conf['namespace']
+        token = conf['token']
+        roster = conf['roster']
+        semester = conf['semester']
+
         count = 0
-        for student in conf['roster']:
-            if args.section and student['section'] != args.section:
+        for student in roster:
+            username = student['username']
+            s_section = student['section']
+
+            if section and s_section != section:
```

Comment:
This should be un-hacked similar to the list expression above. Also, same comment wrt `s_section`.